x/superfluid/keeper/internal/events/emit.go

```diff
@@ -114,6 +114,23 @@ func newSuperfluidUnbondLockEvent(lockId uint64) sdk.Event {
 	)
 }
 
+func EmitSuperfluidUndelegateAndUnbondLockEvent(ctx sdk.Context, lockId uint64) {
+	if ctx.EventManager() == nil {
+		return
+	}
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		newSuperfluidUndelegateAndUnbondLockEvent(lockId),
+	})
+}
+
+func newSuperfluidUndelegateAndUnbondLockEvent(lockId uint64) sdk.Event {
+	return sdk.NewEvent(
+		types.TypeEvtSuperfluidUndelegateAndUnbondLock,
+		sdk.NewAttribute(types.AttributeLockId, fmt.Sprintf("%d", lockId)),
```

Comment:
Out of curiosity: wouldn't you want to emit the amount of tokens undelegated too?

Reply:
I think the event is already emitted by the lock up module
https://github.com/osmosis-labs/osmosis/blob/d28c7412c8d16ed48a20f7b770185d238d83f12b/x/lockup/keeper/lock.go#L247-L249